modules/dnn/src/layers/elementwise_layers.cpp

```diff
@@ -1131,15 +1195,40 @@ struct MishFunctor : public BaseDefaultFunctor<MishFunctor>
 
     inline float calculate(float x) const
     {
-        // Use fast approximation introduced in https://github.com/opencv/opencv/pull/17200
-        if (x >= 8.f)
-        {
-            return x;
-        }
+        float y = x > MISH_THRESHOLD ? std::exp(-x) : 1.f;
+        return x * (1 + 2 * y) / (1 + 2 * y + 2 * y * y);
+    }
+
+    void apply(const float* srcptr, float* dstptr, int stripeStart, int len, size_t planeSize, int cn0, int cn1) const {
+        CV_UNUSED(stripeStart);
+        for (int cn = cn0; cn < cn1; cn++, srcptr += planeSize, dstptr += planeSize) {
+            int i = 0;
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+            v_float32 v_threshold = vx_setall_f32(MISH_THRESHOLD), one = vx_setall_f32(1.f), z = vx_setzero_f32();
+            for (; i <= len - vlanes; i += vlanes) {
+                if (i + vlanes > len) {
+                    if (i == 0 || i == len) {
+                        break;
+                    }
+                    i = len - vlanes;
+                }
```

Comment:
Looks like the extra conditions are redundant. You can replace for loop with:
```
for (; i < len - vlanes; i += vlanes)
```
It's applicable for all cases in the PR.

Reply:
The idea is to also apply intrinsics on the tail by fetching the last vlanes elements, although it may introduce duplicated computation. This trick is borrowed from Halide. Below is an example.

Assume vlanes=4 and we have 7 elements. Without this trick, the last 3 elements have to be processed by a extra for loop. With this trick, it works like two times of loop, both of which process 4 elements, but the second loop takes elements of index 4567, while the first takes 1234 (index 4 element is processed twice).

Reply:
I got the point. The problem is that you introduce several extra conditions in the loop. I propose to extract the condition and add vectorized tail computation with overlap after the loop.

Reply:
Need to talk to @vpisarev because it is his idea to put `if` checks into the loops. Problem is spreading over codebase.

Reply:
> Problem is spreading over codebase.

I volunteer to take care of this if there is a list of this problem.

Reply:
Discussed with @vpisarev  and we decided to drop halide trick for now since elementwise layer is operated in-place. Also removing implementations that causes performance degradation (elu, selu).